src/main/java/org/jabref/logic/importer/AuthorListParser.java

```diff
@@ -209,6 +180,100 @@ public AuthorList parse(@NonNull String listOfNames) {
+        Map<String, String> nameParts = new HashMap<>();
+        Matcher matcher = Pattern.compile("(\\w+)\\s*=\\s*([^,]+)(?:,\\s*|$)").matcher(authorString);
+        while (matcher.find()) {
+            nameParts.put(matcher.group(1).trim(), matcher.group(2).trim());
+        }
+
+        if (nameParts.isEmpty()) {
+            return Optional.empty();
+        }
+
+        String familyName = nameParts.get("family");
+        String givenName = nameParts.get("given");
+        String namePrefix = nameParts.get("prefix");
+        String nameSuffix = nameParts.get("suffix");
+        String usePrefix = nameParts.get("useprefix");
+
+        // handle useprefix=false
+        if ("false".equalsIgnoreCase(usePrefix)) {
+            namePrefix = null;
+        }
+
+        // abbreviate given name
+        String givenNameAbbreviated = abbreviateGivenName(givenName);
+
+        // create Author object
+        return Optional.of(new Author(givenName, givenNameAbbreviated, namePrefix, familyName, nameSuffix));
+    }
+
+
+    /**
+     * Abbreviates the given name by taking the first letter of each word and appending a dot.
+     * Handles cases where the given name is already in an abbreviated format.
+     *
+     * @param givenName the given name string to abbreviate
+     * @return the abbreviated version of the given name
+     */
+    private String abbreviateGivenName(String givenName) {
+        if (givenName == null || givenName.isEmpty()) {
+            return null;
+        }
+        String[] parts = givenName.trim().split("\\s+");
+        StringBuilder abbreviated = new StringBuilder();
+        for (int i = 0; i < parts.length; i++) {
+            if (!parts[i].isEmpty()) {
+                // check if the part is already an initial with a dot
+                if (parts[i].matches("[A-Z]\\.")) {
+                    abbreviated.append(parts[i]);
+                } else {
+                    abbreviated.append(parts[i].charAt(0)).append('.');
+                }
+                if (i < parts.length - 1) {
+                    abbreviated.append(' ');
+                }
+            }
+        }
+        return abbreviated.toString();
+    }
+
+    /**
+     * Splits the author list into individual authors
+     *
+     * @param authorList the author list string
+     * @return a list of individual author strings
+     */
+    private List<String> splitAuthors(String authorList) {
+        List<String> authors = new ArrayList<>();
+        int bracesLevel = 0;
+        int start = 0;
+        for (int i = 0; i < authorList.length(); i++) {
+            char c = authorList.charAt(i);
+            if (c == '{') {
+                bracesLevel++;
+            } else if (c == '}') {
+                bracesLevel--;
+            } else if (i <= authorList.length() - 5 && authorList.substring(i, i + 5).equals(" and ") && bracesLevel == 0) {
+                authors.add(authorList.substring(start, i));
+                i += 4;
```

Comment:
A lot of magic values here, what do the 5 and 4 stand for here? If possible please delcare a constant that has a good name that explains the purpose of these values.